### ISSUE TITLE
Simplified `network` config

### ISSUE THREAD

**Issue by erik-am** [flaviostutz/cdk-practical-constructs#11]
Hi!

I saw the `network` prop that makes it easier to populate the VPC and the subnets for the Lambda.

What I disliked is that I had to supply more properties than vpcID and subnetIDs, but also the route table and availability zones.

I found out a quicker way to achieve the same, by just using the default CDK lambda props, instead of the `network` prop.

Example:

```ts
const vpc = ec2.Vpc.fromLookup(this, 'VPC', {
  vpcId: 'vpc-xxxxxxxxxxxxxxxxx',
});

const myLambda = new BaseNodeJsFunction(
  this,
  'MyLambda',
  {
    /*
        other props
    */
    vpc: vpc,
    vpcSubnets: {
      subnetFilters: [
        ec2.SubnetFilter.byIds([
          'subnet-aaaaaaaa',
          'subnet-bbbbbbbb',
          'subnet-cccccccc',
        ]),
      ],
    },
  },
);
```

Now I only have to know the vpcId and the subnet IDs.

@flaviostutz What do you think of this approach?

**Comment by erik-am** [comment]
To make this work, a bugfix is needed: #12

**Comment by flaviostutz** [comment]
Yes, this is the ”api” way, but what I didn’t like is that everytime you run “synth” then you would need to have your shell configured to access the correct AWS account to do the query (actually that’s why it doesn’t require the other info, because it fetches them with the sdk api).As the account configuration is something we do only once during the initial configuration of the monorepo, I preferred to add more information there once (so the build won’t need any connection to the AWS account and would be more repeatable) than sharing more confusion with all devs everytime they would run the build for any service.  There is a way of making cdk cache it also on dev machine, but I felt it would make things even more confusing to the regular dev because then sometimes it would require the connection and sometimes don’t. We can take a look at https://docs.aws.amazon.com/cdk/v2/guide/context.html and we can spike if it would be a better alternative to our approach, but I think we would then employ a more sofisticated mechanism for something that could be more solved more directly with the way we are approaching it.Also the dev would need to take care of connecting to the correct AWS account with the AWS sdk according to the environment/stage selected, and if he/she does it wrong a generic “network not found” message is returned (because it will try to fetch info for a vpcId that doesn’t exist in the connected account).We already need to add the vpcId and subnet ids anyway, so by adding a couple of more info (yes, I felt it was a bit annoying too!), we make all the builds “offline”, which is something CDK also preaches about too :)What do you think?Flávio Stutz(Sent from the Alps 🤩😂)Sent from my iPhoneOn 30 Apr 2024, at 16:09, Erik Ammerlaan ***@***.***> wrote:﻿
Hi!
I saw the network prop that makes it easier to populate the VPC and the subnets for the Lambda.
What I disliked is that I had to supply more properties than vpcID and subnetIDs, but also the route table and availability zones.
I found out a quicker way to achieve the same, by just using the default CDK lambda props, instead of the network prop.
Example:
const vpc = ec2.Vpc.fromLookup(this, 'VPC', {
  vpcId: 'vpc-xxxxxxxxxxxxxxxxx',
});

const myLambda = new BaseNodeJsFunction(
  this,
  'MyLambda',
  {
    /*
        other props
    */
    vpc: vpc,
    vpcSubnets: {
      subnetFilters: [
        ec2.SubnetFilter.byIds([
          'subnet-aaaaaaaa',
          'subnet-bbbbbbbb',
          'subnet-cccccccc',
        ]),
      ],
    },
  },
);
Now I only have to know the vpcId and the subnet IDs.
@flaviostutz What do you think of this approach?

**Comment by flaviostutz** [comment]
Yes, this is the ”api” way, but what I didn’t like is that everytime you run “synth” then you would need to have your shell configured to access the correct AWS account to do the query (actually that’s why it doesn’t require the other info, because it fetches them with the sdk api).

As the account configuration is something we do only once during the initial configuration of the monorepo, I preferred to add more information there once (so the build won’t need any connection to the AWS account and would be more repeatable) than sharing more confusion with all devs everytime they would run the build for any service.

**Comment by erik-am** [comment]
Thanks for explaining more about the rationale!

I definitely see the value now. It's nice if it doesn't have to resolve the value with a live connection.
On the other hand, I'm also not that fond of all the "magic strings" with configuration (even though it's set up once), and it's nice (from a programming point of view) if you can use the `ec2.Vpc` object.

Maybe it depends whether or not you are already using other stuff that requires a connection anyway, then it wouldn't matter that much. If this is the only thing that needs a connection, then I agree, the `network` option is nicer.

At least it's good if people can choose between the two methods. 🙂 
I'll experiment with both options and see where it leads me. 😀

**Comment by flaviostutz** [comment]
Nice! I couldn't find other use cases in which I would need the connection, that's why I primarily chose this method. But I agree with enabling both options. 

Could you write an example of using both options for the readme file after you experiment with it?
